### PR TITLE
chore: use Go 1.16.5

### PR DESCRIPTION
**Pull request by AlekSi**
Bump tools, pkgs, extras.

### CODE REVIEW

**Review by smira — APPROVED** [review]

**Review by Unix4ever — DISMISSED** [review]

**Review by smira — DISMISSED** [review]

File: pkg/machinery/constants/constants.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 
 const (
 	// DefaultKernelVersion is the default Linux kernel version.
-	DefaultKernelVersion = "5.10.29-talos"
+	DefaultKernelVersion = "5.10.38-talos"
```
Comment (AlekSi) [inline pkg/machinery/constants/constants.go:16]: Should it be added to release.toml now or later? What is our process for that?

Comment (AlekSi) [reply pkg/machinery/constants/constants.go:16]: And should we update it to 5.10.42?

Comment (smira) [reply pkg/machinery/constants/constants.go:16]: we can update it later

**Review by smira — APPROVED** [review]

**Pull request merged**